File: algorithms/recursion/python/pascal_triangle_2.py
def get_row(row_index):
    if row_index == 0:
        return [1]

    if row_index == 1:
        return [1, 1]

    result = []
    result.append(1)

    previous_result = get_row(row_index - 1)

    for index in range(1, row_index):
        result.append(previous_result[index - 1] + previous_result[index])

    result.append(1)
    return result


def main():
    print(get_row(3))


if __name__ == "__main__":
    main()

File: algorithms/recursion/java/PascalTriangle2.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class PascalTriangle2 {
    public static List<Integer> getRow(int rowIndex) {
        List<Integer> result = new ArrayList<>();
        if (rowIndex == 0) {
            return Arrays.asList(1);
        }

        if (rowIndex == 1) {
            return Arrays.asList(1, 1);
        }

        result.add(1);

        List<Integer> previous = getRow(rowIndex - 1);
        for (int i = 1; i < rowIndex; i++) {
            result.add(previous.get(i - 1) + previous.get(i));
        }

        result.add(1);
        return result;
    }

    public static void main(String[] args) {
        List<Integer> result = getRow(3);

        result.forEach(System.out::println);
    }
}
